WIP - Add workflow for adoc to html (#1033)
hazelcast/hz-docs: docs/modules/whats-new/whats-new.adoc

What's new in Platform 5.4

> First of all, starting with this release of Hazelcast Platform, the minimum supported Java version has become 17.

Reviewer: ```suggestion
From this release of Hazelcast Platform, the minimum supported Java version is 17.
```

Reviewer: Perhaps add something about the benefits of this (better CPU use, additional language features, security updates, and improved support for dev tools (IDEs)

> You can now deploy, undeploy or redeploy your custom classes by using the user code name spacing feature. The feature provides a container for Java classpath resources, such as your classes, and accompanying artifacts like property files - enabling namespace isolation to ensure that access to resources in different namespaces can be managed through configuration.

Reviewer: ```suggestion
You can now deploy, or redeploy your custom classes by using the user code name spacing feature. The feature provides a container for Java classpath resources, such as your classes, and accompanying artifacts like property files - enabling namespace isolation to ensure that access to resources in different namespaces can be managed through configuration.
```

Reviewer: There was a late change that meant you cannot undeploy at runtime, the agreement was to focus on deploying and redeploying

> We've also added a map data structure aligned with your CP subsystem - (C)onsistent and (P)artition tolerant: CPMap. It is a distributed implementation of a minimal key-value interface supporting many atomic operations like get, delete and put.

Reviewer: ```suggestion
We've also added a map data structure aligned with your CP subsystem - (C)onsistent and (P)artition tolerant: CPMap. This distributed implementation of a minimal key-value interface supports many atomic operations such as get, delete and put.
```
